tests/FeeSharingCollectorTest.js

```diff
@@ -291,21 +288,508 @@ contract("FeeSharingCollectorProxy:", (accounts) => {
+            expect([
+                nextCheckpoint.checkpointNum.toNumber(),
+                nextCheckpoint.hasSkippedCheckpoints,
+                nextCheckpoint.hasFees,
+            ]).to.eql([10, true, true]);
+        });
+        it("getNextPositiveUserCheckpoint for RBTC returns correct [checkpointNum, hasSkippedCheckpoints, hasFees]", async () => {
+            feeSharingCollector = await FeeSharingCollectorMockup.new(
+                sovryn.address,
+                staking.address
+            );
+            await sovryn.setFeesController(feeSharingCollector.address);
+
+            let nextCheckpoint = await feeSharingCollector.getNextPositiveUserCheckpoint(
+                account1,
+                RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT
+            );
+            expect([
+                nextCheckpoint.checkpointNum.toNumber(),
+                nextCheckpoint.hasSkippedCheckpoints,
+                nextCheckpoint.hasFees,
+            ]).to.eql([0, false, false]);
+
+            await stake(900, root);
+            const userStake = 100;
+
+            await createCheckpoints(9);
+
+            nextCheckpoint = await feeSharingCollector.getNextPositiveUserCheckpoint(
+                account1,
+                RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT
+            );
+            expect([
+                nextCheckpoint.checkpointNum.toNumber(),
+                nextCheckpoint.hasSkippedCheckpoints,
+                nextCheckpoint.hasFees,
+            ]).to.eql([9, true, false]);
+
+            await feeSharingCollector.setUserProcessedCheckpoints(
+                account1,
+                RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT,
+                9
+            );
+
+            nextCheckpoint = await feeSharingCollector.getNextPositiveUserCheckpoint(
+                account1,
+                RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT
+            );
+            expect([
+                nextCheckpoint.checkpointNum.toNumber(),
+                nextCheckpoint.hasSkippedCheckpoints,
+                nextCheckpoint.hasFees,
+            ]).to.eql([9, false, false]);
+
+            // undo mock user processed checkpoints
+            await feeSharingCollector.setUserProcessedCheckpoints(
+                account1,
+                RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT,
+                0
+            );
+
+            await SOVToken.transfer(account1, userStake);
+            await stake(userStake, account1);
+            await createCheckpoints(1);
+
+            nextCheckpoint = await feeSharingCollector.getNextPositiveUserCheckpoint(
+                account1,
+                RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT
+            );
+            expect([
+                nextCheckpoint.checkpointNum.toNumber(),
+                nextCheckpoint.hasSkippedCheckpoints,
+                nextCheckpoint.hasFees,
+            ]).to.eql([10, true, true]);
+        });
+    });
 
     describe("FeeSharingCollectorProxy", () => {
+        /*before(async () => {
```

Comment:
Why is this commented out?

Reply:
a leftover after experiments with restoring snapshots  
not necessary here as restored in the upfollowing tests as needed but uncommented to keep it uniform